src/main/native/io_sqooba_traildb_TrailDBNative.cpp

```diff
@@ -482,35 +479,44 @@ JNIEXPORT jint JNICALL Java_io_sqooba_traildb_TrailDBNative_tdbCursorNext
 	tdb_cursor *cursor = (tdb_cursor*) env->GetDirectBufferAddress(jcursor);
 
 	// Call lib.
-	const tdb_event *event = tdb_cursor_next(cursor);
+	const tdb_event *event;
+	event = tdb_cursor_next(cursor);
 
 	// Check if there is no more events.
 	if(!event) {
 		return -1;
 	}
 
-	// Get struct elements.
-	uint64_t timestamp = event->timestamp;
-	uint64_t num_items = event->num_items;
-	const tdb_item *items_ptr = event->items;
-
+	// Store field in TrailDBEvent.
+	env->SetLongField(jevent, JFID_traildbEvent_timestamp, event->timestamp);
+	env->SetLongField(jevent, JFID_traildbEvent_numItems, event->num_items);
+	env->SetLongField(jevent, JFID_traildbEvent_items, (long)event->items);
 
-  	jlongArray newArray = env->NewLongArray(num_items);
-    jlong *narr = env->GetLongArrayElements(newArray, NULL);
+	return 0;
+}
 
-	unsigned int i;
-    for (i = 0; i < num_items; i++) {
-        narr[i] = items_ptr[i];
-    }
+JNIEXPORT jstring JNICALL Java_io_sqooba_traildb_TrailDBNative_eventGetItemValue
+  (JNIEnv *env, jobject thisObject, jobject jdb, jint jindex, jobject jvalueLength, jobject jevent) 
+{
+	
+	// Convert arguments.
+	const tdb *db = (tdb*) env->GetDirectBufferAddress(jdb);
+	uint64_t value_length;
 
-	env->SetLongField(jevent, JFID_traildbEvent_timestamp, timestamp);
-	env->SetLongField(jevent, JFID_traildbEvent_numItems, num_items);
-	env->SetObjectField(jevent, JFID_traildbEvent_items, newArray);
+	jclass jc = env->GetObjectClass(jvalueLength);
+	jmethodID mid = env->GetMethodID(jc, "putLong","(J)Ljava/nio/ByteBuffer;");
 
-	env->ReleaseLongArrayElements(newArray, narr, 0);
+	// Get the items pointer.
+	const tdb_item *items;
+	items = (tdb_item *) env->GetLongField(jevent, JFID_traildbEvent_items);
+	
+	// Call lib.
+	const char* v = tdb_get_item_value(db, items[jindex], &value_length);
 
-	return 0;
+	// Store to the buffer the what has been put in the pointer.
+	env->CallObjectMethod(jvalueLength, mid, value_length);
```

Comment:
what buffer ? from which pointer ? 
if I understand correctly, jvalueLength is the buffer, but the pointer... v ?

Reply:
The buffer is the ByteBuffer jvalueLength which will hold the size of the String returned by tdb_get_item_value.
The pointer was value_length because it was a uint64_t* previously but this comment is not up to date.
As said in the previous comment, I will get rid of all this ByteBuffer stuff soon.